src/templates.rs

```diff
@@ -339,6 +367,41 @@ pub static TEMPLATE_BWD_COMPILATION_METRICS: &str = r#"
     {{ else }}
     <p> No failures! </p>
     {{ endif }}
+    "#,
+    TEMPLATE_QUERY_PARAM_SCRIPT,
+    r#"
 </body>
 </html>
-"#;
+"#
+);
+
+pub const TEMPLATE_QUERY_PARAM_SCRIPT: &str = r#"
+    <script>
+    document.addEventListener('DOMContentLoaded', function() \{
+
+        // Append the current URL's query parameters to all relative links on the page
+        const queryParams = new URLSearchParams(window.location.search);
+        function appendQueryParams(url) \{
+            if (queryParams.size === 0) return url; // No query params, return original URL
+
+            const parsedUrl = new URL(url, window.location.href);
```

Comment:
This isn't right. We actually use the `<base>` tag on some internal pages and this will set the wrong base. Why do you need the Base anyway, it's not like URL requires it.

New test plan: Please check internal links from compilation_metrics.html also work

Reply:
Fixed.